apiserver/remoterelations/remoterelations.go

```diff
@@ -184,55 +184,134 @@ func (api *RemoteRelationsAPI) RelationUnitSettings(relationUnits params.Relatio
+		// required CIDRs for which we need to allow ingress to this model.
+		sourceSt, closer, err := api.pool.Get(remoteRelation.SourceModelUUID)
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
-		result := &params.RemoteRelation{
-			Id:   rel.Id(),
-			Life: params.Life(rel.Life().String()),
-			Key:  tag.Id(),
+		defer closer()
+
+		// TODO(wallyworld) - as a first implementation, just get all CIDRs from the model
+		all, err := sourceSt.AllIPAddresses()
+		if err != nil {
+			return nil, errors.Trace(err)
 		}
-		for _, ep := range rel.Endpoints() {
-			// Try looking up the info for the remote application.
-			_, err := api.st.RemoteApplication(ep.ApplicationName)
-			if err != nil && !errors.IsNotFound(err) {
+
+		for _, addr := range all {
+			ip, _, err := net.ParseCIDR(addr.SubnetCIDR())
+			if err != nil {
 				return nil, errors.Trace(err)
-			} else if err == nil {
-				result.RemoteEndpointName = ep.Name
+			}
+			if ip.IsLoopback() || ip.IsMulticast() {
 				continue
 			}
-			// Try looking up the info for the local application.
-			_, err = api.st.Application(ep.ApplicationName)
-			if err != nil && !errors.IsNotFound(err) {
-				return nil, errors.Trace(err)
-			} else if err == nil {
-				result.ApplicationName = ep.ApplicationName
-				result.Endpoint = params.RemoteEndpoint{
-					Name:      ep.Name,
-					Interface: ep.Interface,
-					Role:      ep.Role,
-					Scope:     ep.Scope,
-					Limit:     ep.Limit,
-				}
+			// TODO(walluworld) - We only support IPv4 addresses as not all providers support IPv6.
+			if ip.To4() == nil {
 				continue
 			}
+			result.CIDRs = append(result.CIDRs, addr.SubnetCIDR())
+		}
+		logger.Debugf("Ingress CIDRS for remote relation %v from model %v: %v", relTag, remoteRelation.SourceModelUUID, result.CIDRs)
+		return &result, nil
+	}
+
+	for i, entity := range entities.Entities {
+		networks, err := one(entity)
+		if err != nil {
+			results.Results[i].Error = common.ServerError(err)
+			continue
 		}
-		return result, nil
+		results.Results[i].Result = networks
+	}
+	return results, nil
+}
+
+func (api *RemoteRelationsAPI) remoteRelation(entity params.Entity) (*params.RemoteRelation, error) {
+	tag, err := names.ParseRelationTag(entity.Tag)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	rel, err := api.st.KeyRelation(tag.Id())
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	result := &params.RemoteRelation{
+		Id:   rel.Id(),
+		Life: params.Life(rel.Life().String()),
+		Key:  tag.Id(),
+	}
+	for _, ep := range rel.Endpoints() {
+		// Try looking up the info for the remote application.
+		remoteApp, err := api.st.RemoteApplication(ep.ApplicationName)
+		if err != nil && !errors.IsNotFound(err) {
+			return nil, errors.Trace(err)
+		} else if err == nil {
+			result.RemoteEndpointName = ep.Name
+			result.SourceModelUUID = remoteApp.SourceModel().Id()
+			continue
+		}
+		// Try looking up the info for the local application.
```

Comment:
This is existing code just extracted intoa method. The TR;DR; is that relation endpoint application names may refer to either a remote or local application. Both these collections are in the consuming model. The names need to be unique. In the offering model, there's also a corresponding application to the remote proxy in this model; those names don't need to match.